pkg/trillianclient/trillian_client.go

```diff
@@ -75,18 +185,170 @@ func unmarshalLogRoot(logRoot []byte) (types.LogRootV1, error) {
 	return root, nil
 }
 
-func (t *TrillianClient) root(ctx context.Context) (types.LogRootV1, error) {
-	rqst := &trillian.GetLatestSignedLogRootRequest{
-		LogId: t.logID,
-	}
-	resp, err := t.client.GetLatestSignedLogRoot(ctx, rqst)
+// ensureStarted initializes the shared LogClient and starts the updater once.
+//
+// Locking strategy: this uses a double-checked initialization gate so network
+// calls do not hold the mutex. Only one goroutine performs the initial RPCs,
+// others wait on the condition variable until initialization completes. This
+// avoids head-of-line blocking while keeping state updates atomic.
+func (t *TrillianClient) ensureStarted(ctx context.Context) error {
```

Comment:
Could you have this called when initializing the client, rather than from each client function?

Other option for handling concurrency, have you looked into `sync.OnceFunc` to ensure this runs only once?